pkg/zrouter/zrouter.go

```diff
@@ -247,6 +277,55 @@ func (r *zrouter) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	r.router.ServeHTTP(w, req)
 }
 
+func LogTopJWTPathMetrics(ctx context.Context, zCache zcache.RemoteCache, updateInterval time.Duration, topN int) {
+	if updateInterval == 0 {
+		updateInterval = defaultUpdateInterval
+	}
+	ticker := time.NewTicker(updateInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			func() {
+				defer func() {
+					if r := recover(); r != nil {
+						logger.GetLoggerFromContext(ctx).Errorf("Recovered in LogTopJWTPathMetrics: %v", r)
+					}
+				}()
+
+				topTokens, err := zCache.ZRevRangeWithScores(ctx, zmiddlewares.PathUsageByJWTKey, 0, int64(topN-1))
+				if err != nil {
+					logger.GetLoggerFromContext(ctx).Errorf("Error fetching top tokens from cache: %v", err)
+					return
+				}
+
+				for _, item := range topTokens {
+					metricKey, ok := item.Member.(string)
+					if !ok {
+						logger.GetLoggerFromContext(ctx).Errorf("Failed to type assert member to string: %v", item.Member)
+						continue
+					}
+
+					parts := strings.Split(metricKey, ":")
+					if len(parts) != 2 {
+						logger.GetLoggerFromContext(ctx).Errorf("Unexpected metric key format: %v", metricKey)
+						continue
+					}
+
+					jti, path := parts[0], parts[1]
+					count := item.Score
+
+					logger.GetLoggerFromContext(ctx).Infof("JWT ID: %s, Path: %s, Count: %f", jti, path, count)
```

Comment:
What if we actually log everything together? We won't get spared lines in the log so.